src/main/java/com/vinylteam/vinyl/util/RawOffersSorter.java

```diff
@@ -46,51 +46,68 @@
-        if (rawOffers != null && uniqueVinyl != null && offers != null) {
-            int percentMatching = 75;
-            String uniqueVinylRelease = uniqueVinyl.getRelease();
-            char lastCharInRawRelease = uniqueVinylRelease.charAt(uniqueVinylRelease.length() - 1);
-            if (lastCharInRawRelease > '0' && lastCharInRawRelease < '9') {
-                percentMatching = 90;
-            }
-            String[] preparedFullNameForMatching = Arrays.stream(uniqueVinyl.getFullName().split("[- ()!@$%^&*_={}:;\"']")).filter(e -> e.trim().length() > 0).toArray(String[]::new);
-            Iterator<RawOffer> rawOfferIterator = rawOffers.iterator();
-            uniqueVinyl.setHasOffers(false);
-            while (rawOfferIterator.hasNext()) {
-                RawOffer rawOffer = rawOfferIterator.next();
-                String rawOfferFullName = rawOffer.getArtist() + " - " + rawOffer.getRelease();
-                int currentMatching = 0;
-                for (String prepareItem : preparedFullNameForMatching) {
-                    if (rawOfferFullName.toLowerCase().contains(prepareItem.toLowerCase())) {
-                        currentMatching++;
-                    }
-                }
-                if (Objects.equals(getParametersForComparison(uniqueVinyl.getRelease()), getParametersForComparison(rawOffer.getRelease())) &&
-                        Objects.equals(getParametersForComparison(uniqueVinyl.getArtist()), getParametersForComparison(rawOffer.getArtist()))) {
-                    if (((float) currentMatching) / preparedFullNameForMatching.length * 100 > percentMatching) {
-                        Offer offer = new Offer();
-                        offer.setUniqueVinylId(uniqueVinyl.getId());
-                        offer.setShopId(rawOffer.getShopId());
-                        offer.setPrice(rawOffer.getPrice());
-                        offer.setCurrency(rawOffer.getCurrency());
-                        offer.setGenre(rawOffer.getGenre());
-                        offer.setCatNumber(rawOffer.getCatNumber());
-                        offer.setInStock(rawOffer.isInStock());
-                        offer.setOfferLink(rawOffer.getOfferLink());
-                        offers.add(offer);
-                        uniqueVinyl.setHasOffers(true);
-                        log.debug("Added new offer {'offer':{}}", offer);
-                        rawOfferIterator.remove();
-                    }
-                }
-            }
-            uniqueVinyl.setHasOffers(!offers.isEmpty());
-        } else {
+        if (rawOffers == null || uniqueVinyl == null || offers == null) {
             RuntimeException e = new NullPointerException();
             log.error("At least one of passed arguments is null {'rawOffers':{}, 'uniqueVinyl':{}, 'offers':{}}",
                     rawOffers, uniqueVinyl, offers, e);
             throw e;
         }
```

Comment:
I think it is not good idea to throw NullPointerException

Reply:
Actually this check is redundant at all, no chance to pass null in any of parameters - collections are initialized before in program flow, uniqueVinyl is taken from iterator, and in case iterator does not have any more vinyls, the special logic adds new one - to be reworked